doc/developer-guides/coding_guidelines.rst

```diff
@@ -1288,3 +1901,40 @@ Compliant example::
        }
 
 
+ID-06: The typedef name shall be unique
+=======================================
+
+The typedef name shall be unique and not be used for any other purpose.
+
+Compliant example::
+
+    typedef unsigned int uint32_t;
+    
+    uint32_t showcase;
+
+.. rst-class:: non-compliant-code
+
+   Non-compliant example::
+
+       typedef unsigned int uint32_t;
+       
+       uint32_t uint32_t;
+
+
+ID-07: Name defined by developers shall not start with underscore
+=================================================================
+
+Name defined by developers shall not start with underscore since it is reserved
```

Comment:
A better explanation would be:
```
All names starting with one or two underscores are reserved for use
by the compiler and standard libraries to eliminate potential conflicts
with user-defined names.
```

Reply:
Yes, it is much better. I will update it in next version.